### PR TITLE
Centralize and improve system-probe config

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

- uses viper for env var binding and defaults
- improves module enable/disable logic
- centralizes config so there is one place to declare a config value, and one place to read it.
- remove process-agent config as an intermediary for system-probe config.

### Motivation

- Bugs and mismatch between config `SetKnown`, config reading to process-agent, and config reading in system-probe.
- Inconsistency of how system-probe handled config versus other agents. 
- lack of environment variable bindings for most of system-probe config options.

### Additional Notes

- I fixed a bug in some process-agent config tests where it would wait 60s on each test for a gRPC connection to timeout. This wasn't happening before because it wasn't getting the default address value of `localhost` for IPC. See https://github.com/DataDog/datadog-agent/commit/e0462c90470ef367df7bfa70f9bd8378dcf79f49

### Describe your test plan

Automated tests cover most of the settings, but feel free to change a config setting and ensure the new value is used by system-probe.

### CODE REVIEW

**Review by heyronhay — APPROVED** [review]
So much nicer, thanks for doing this!

File: cmd/system-probe/config/config.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+	if cfg.GetBool(key(spNS, "enable_tcp_queue_length")) {
+		log.Info("system_probe_config.enable_tcp_queue_length detected, will enable system-probe with TCP queue length check")
+		c.EnabledModules[TCPQueueLengthTracerModule] = struct{}{}
+	}
+	if cfg.GetBool(key(spNS, "enable_oom_kill")) {
+		log.Info("system_probe_config.enable_oom_kill detected, will enable system-probe with OOM Kill check")
+		c.EnabledModules[OOMKillProbeModule] = struct{}{}
+	}
+	if cfg.GetBool("runtime_security_config.enabled") || cfg.GetBool("runtime_security_config.fim_enabled") {
```
Comment (safchain) [inline cmd/system-probe/config/config.go:173]: I would move this logic in each module, it will decouple the modules a bit more. What about adding a `IsEnabled(*Config) bool` function in the `Factory` struct ? Probably something we could do in another PR.

Comment (brycekahle) [reply cmd/system-probe/config/config.go:173]: That would create an import loop, because then the config would import the modules, which import the config.

File: pkg/config/system_probe.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	cfg.BindEnvAndSetDefault(join(spNS, "log_file"), defaultSystemProbeLogFilePath)
+	cfg.BindEnvAndSetDefault(join(spNS, "log_level"), "info", "DD_LOG_LEVEL", "LOG_LEVEL")
+	cfg.BindEnvAndSetDefault(join(spNS, "debug_port"), 0)
+
+	cfg.BindEnvAndSetDefault(join(spNS, "dogstatsd_host"), "127.0.0.1")
+	cfg.BindEnvAndSetDefault(join(spNS, "dogstatsd_port"), 8125)
+
+	cfg.BindEnvAndSetDefault(join(spNS, "profiling.enabled"), false, "DD_SYSTEM_PROBE_PROFILING_ENABLED")
+	cfg.BindEnvAndSetDefault(join(spNS, "profiling.site"), DefaultSite, "DD_SYSTEM_PROBE_PROFILING_SITE", "DD_SITE")
+	cfg.BindEnvAndSetDefault(join(spNS, "profiling.profile_dd_url"), "", "DD_SYSTEM_PROBE_PROFILING_DD_URL", "DD_APM_PROFILING_DD_URL")
```
Comment (ogaca-dd) [inline pkg/config/system_probe.go:39]: `DD_SYSTEM_PROBE_PROFILING_DD_URL` seems to be a new env var and does not match `profiling.profile_dd_url`. Is it expected?

**Review by ogaca-dd — APPROVED** [review]
LGTM on Agent Core side.
I reviewed only the changes in `pkg/config/*`

File: pkg/config/system_probe.go (hunk 2 above)
Comment (brycekahle) [reply pkg/config/system_probe.go:39]: I was matching the format of `DD_APM_PROFILING_DD_URL` but specific to system-probe. Happy to change it, if we think something else is better.

**Review by shang-wang — APPROVED** [review]

**Review by p-lambert — APPROVED** [review]

**Review by safchain — APPROVED** [review]
